merkle/hashers/tree_hasher.go

```diff
@@ -93,7 +137,18 @@ func NewLogHasher(h trillian.HashStrategy) (LogHasher, error) {
 	return nil, fmt.Errorf("LogHasher(%s) is an unknown hasher", h)
 }
 
+// NewInplaceLogHasher returns an InplaceLogHasher.
```

Comment:
No. It does create new instances.

Reply:
It takes one from a hashmap, doesn't it?

Reply:
Oh wait, I'm wrong

Reply:
Yeah the others happened to be stateless so it didn't matter they were shared. But if somebody does register a `LogHasher` that isn't then it will break.